### ISSUE TITLE
About torch.utils.checkpoint

### ISSUE THREAD

**Issue by thought-tts-demo** [haidog-yaqub/MeanFlow#4]
Thanks for your wonderful work!

When I tried to add gradient checkpointing to my own model, I encountered ``Checkpointing is not compatible with .grad() or when an `inputs` parameter is passed to .backward()`` in ``jvp`` calculation. So I would like to ask did you face the same problem? How did you fix it? ❤️🌹
